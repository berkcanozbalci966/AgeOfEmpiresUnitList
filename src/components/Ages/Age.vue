<template>
  <div>
    <span> {{ ageInformation }} </span>
  </div>
</template>

<script>
export default {
  name: "Age",
  props: {
    ageInformation: {
      type: String,
      default: () => "",
    },
  },
};
</script>

<style lang="scss" scoped>
div {
  border: 1px solid black;
  padding: 4px;
  &:hover {
    cursor: pointer;
  }
}
</style>
