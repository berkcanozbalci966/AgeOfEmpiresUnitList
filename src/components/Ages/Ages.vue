<template>
  <div>
    <Age
      v-for="(ageName, index) in ageList"
      :key="index"
      :ageInformation="{ ageName, index, selectedAge }"
      @ageClick="changeSelectedAge($event)"
    />
  </div>
</template>

<script>
import Age from "./Age.vue";
import { mapState, mapMutations } from "vuex";
export default {
  name: "Ages",
  components: { Age: Age },
  computed: {
    ...mapState({
      ageList: (state) => state.Age.ageList,
      selectedAge: (state) => state.Age.selectedAge,
    }),
  },
  methods: {
    setAge(selectedAgeIndex) {
      this.selectedAge = selectedAgeIndex;
    },
    ...mapMutations("Age", ["changeSelectedAge"]),
  },
};
</script>

<style lang="scss" scoped>
div {
  display: flex;
}
</style>
