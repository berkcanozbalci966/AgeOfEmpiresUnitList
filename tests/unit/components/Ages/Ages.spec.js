import { mount } from "@vue/test-utils";
import Vuex from "vuex";
import Ages from "../../../../src/components/Ages/Ages";
import AgeModule from "../../../../src/store/modules/age";

describe("Ages", () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        Age: AgeModule,
      },
    });
  });

  it("list render", () => {
    const component = mount(Ages, {
      global: {
        plugins: [store],
      },
    });

    expect(component.html()).toContain("All");
    expect(component.html()).toContain("Dark");
    expect(component.html()).toContain("Feudal");
    expect(component.html()).toContain("Castle");
    expect(component.html()).toContain("Imperial");
  });

  it("Render correctly", () => {
    const component = mount(Ages, {
      global: {
        plugins: [store],
      },
    });

    expect(component.html()).toMatchSnapshot();
  });

  it("All option is selected", () => {
    const component = mount(Ages, {
      global: {
        plugins: [store],
      },
    });
    expect(component.findAll("div")[1].classes("active")).toBe(true);
  });

  it("Option Length must be 5", () => {
    const component = mount(Ages, {
      global: {
        plugins: [store],
      },
    });
    expect(component.findAll("[data-test='age']")).toHaveLength(5);
  });

  it("Change option correctly", async () => {
    const component = mount(Ages, {
      global: {
        plugins: [store],
      },
    });

    await component.findAll("[data-test='age']")[1].trigger("click");
    expect(component.findAll("[data-test]")[1].classes("active")).toBe(true);
  });
});
