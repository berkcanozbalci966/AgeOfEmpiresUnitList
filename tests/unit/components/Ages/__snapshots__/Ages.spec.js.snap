// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ages Render correctly 1`] = `
<div>
  <div class="active" data-test="age"><span>All</span></div>
  <div class="" data-test="age"><span>Dark</span></div>
  <div class="" data-test="age"><span>Feudal</span></div>
  <div class="" data-test="age"><span>Castle</span></div>
  <div class="" data-test="age"><span>Imperial</span></div>
</div>
`;
